lib/Controller/LoginController.php

```diff
@@ -213,6 +216,10 @@ public function login(int $providerId, string $redirectUrl = null) {
 		$url = $discovery['authorization_endpoint'] . '?' . http_build_query($data);
 		$this->logger->debug('Redirecting user to: ' . $url);
 
+		if ($this->request->isUserAgent(['/Safari/'])) {
```

Comment:
We can probably further only use this condition in cases when there is no redirect happening from https://github.com/nextcloud/server/blob/master/lib/private/AppFramework/Middleware/Security/SecurityMiddleware.php#L268